src/TNetwork.cpp

```diff
@@ -153,7 +151,8 @@ void TNetwork::TCPServerMain() {
             if (ec) {
                 beammp_errorf("failed to accept: {}", ec.message());
             }
-            ClientSocket.set_option(rcv_timeout_option{ 120000 }); //timeout of 120seconds
+            const int timeout = 120000; //timeout of 120seconds
+            ::setsockopt(ClientSocket.native_handle(), SOL_SOCKET, SO_RCVTIMEO, reinterpret_cast<const char*>(&timeout), sizeof timeout);
```

Comment:
`setsockopt` on windows expects a `const char*`, on linux a `const void*`. So far, so good.
Hi, thanks for your contribution!

The WinSock API says the argument should be `milliseconds` (so 120000 is right), whereas the linux api says it expects a `timeval` struct (https://www.man7.org/linux/man-pages/man3/timeval.3type.html) which is a struct with seconds and µs.
Not sure how you tested it, but I don't see that working, since 120000 will be most likely be "understood" as the seconds in the struct, and the µs will be UB to access since there's nothing there.

I'm aware that the current timeout in the code before also doesn't work.

Reply:
'Tested' in the sense that I just compiled it for both platforms and made sure the server started and was joinable. 

I've updated this PR to pass milliseconds to `setsockopt` on windows and a `timeval` struct on other platforms (OS X also expects a timeval when setting `SO_RCVTIMEO`)